cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	}
+
+	if len(config.CfNativeDeployParameters) > 0 {
+		deployStatement = append(deployStatement, strings.FieldsFunc(config.CfNativeDeployParameters, func(c rune) bool {
+			return c == ' '
+		})...)
+	}
+
+	stopOldAppIfRunning := func(_cmd command.ExecRunner) error {
+
+		if config.KeepOldInstance && config.DeployType == "blue-green" {
+			oldAppName := deployConfig.AppName + "-old"
+			_ = oldAppName
+
+			var buff bytes.Buffer
+
+			oldOut := _cmd.Stdout
+			_ = oldOut
+			_cmd.Stdout(&buff)
+
+			defer func() {
+				_cmd.Stdout(log.Writer())
+			}()
+
+			err := _cmd.RunExecutable("cf", "stop", oldAppName)
+
+			if err != nil {
+
+				cfStopLog := buff.String()
+
+				if !strings.Contains(cfStopLog, oldAppName+" not found") {
+					return fmt.Errorf("Could not stop application %s. Error: %s", oldAppName, cfStopLog)
+				}
+				log.Entry().Infof("Cannot stop application '%s': %s", oldAppName, cfStopLog)
+
+			} else {
+				log.Entry().Infof("Old application '%s' has been stopped.", oldAppName)
+			}
+		}
+
+		return nil
+	}
+
+	return cfDeploy(config, deployStatement, additionalEnvironment, stopOldAppIfRunning, cmd)
+}
+
+func getManifest(name string) (cloudfoundry.Manifest, error) {
+	return cloudfoundry.ReadManifest(name)
+}
+
+func getAppNameOrFail(config *cloudFoundryDeployOptions) (string, error) {
+
+	if len(config.AppName) > 0 {
+		return config.AppName, nil
+	}
+	if config.DeployType == "blue-green" {
+		return "", fmt.Errorf("Blue-green plugin requires app name to be passed (see https://github.com/bluemixgaragelondon/cf-blue-green-deploy/issues/27)")
+	}
+	if len(config.Manifest) == 0 {
+		return "", fmt.Errorf("Manifest file not provided in configuration. Cannot retrieve app name")
+	}
+	fileExists, err := fileUtils.FileExists(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	if !fileExists {
+		return "", fmt.Errorf("Manifest file '%s' not found. Cannot retrieve app name", config.Manifest)
+	}
+	m, err := _getManifest(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	apps, err := m.GetApplications()
+	if err != nil {
+		return "", err
+	}
+	if len(apps) > 0 {
+		namePropertyExists, err := m.ApplicationHasProperty(0, "name")
+		if err != nil {
+			return "", err
+		}
+		if namePropertyExists {
+			appName, err := m.GetApplicationProperty(0, "name")
+			if err != nil {
+				return "", err
+			}
+			if name, ok := appName.(string); ok {
+				if len(name) > 0 {
+					return name, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No appName available in manifest '%s'", config.Manifest)
+}
+
+func handleSmokeTestScript(smokeTestScript string) ([]string, error) {
+
+	if smokeTestScript == "blueGreenCheckScript.sh" {
+		// what should we do if there is already a script with the given name? Should we really overwrite ...
+		err := fileUtils.FileWrite(smokeTestScript, []byte(defaultSmokeTestScript), 0755)
+		if err != nil {
+			return []string{}, err
```

Comment:
Maybe good to wrap the error here.
```suggestion
			return []string{}, fmt.Errorf("failed to write default smoke-test script: %w", err)
```